### PR TITLE
feat: add lookupUserByUsernameV2()

### PR DESCRIPTION
**Pull request by eurbs**
- **feat: add lookupUserByUsernameV2()**
    - Retrieves the specified user via their username (if found).
    - Docs: https://docs.neynar.com/reference/user-by-username-v2

### CODE REVIEW

**Review by Shreyaschorge — CHANGES_REQUESTED** [review]
Build is failing
image

File: src/neynar-api/v1/client.ts [hunk 1]
```diff
@@ -73,7 +73,8 @@ export class NeynarV1APIClient {
     if (axiosInstance === undefined) {
       axiosInstance = axios.create({
         headers: {
-          'x-sdk-version': version
+          'x-sdk-version': version,
+          'x-api-key': apiKey,
```
Comment (eurbs) [inline src/neynar-api/v1/client.ts:77]: @Shreyaschorge this change seems to have worked. tests:

```js
const apiKey = '{YOUR API KEY HERE}'
const clientV1 = new NeynarV1APIClient(apiKey, { basePath: 'https://api.neynar.com'});

clientV1.fetchRecentUsers({viewerFid: 3, limit: 1}).then(response => console.log('fetchRecentUsers res:', response));
clientV1.fetchAllCastsLikedByUser(3, {viewerFid: 3, limit: 1}).then(response => console.log('fetchAllCastsLikedByUser res:', response));
clientV1.lookupUserByFid(3, 3).then(response => console.log('fetchAllCastsLikedByUser res:', response));
clientV1.lookupUserByUsername('manan', 3).then(response => console.log('lookupUserByUsername res:', response));
clientV1.lookupCustodyAddressForUser(3).then(response => console.log('lookupCustodyAddressForUser res:', response));
```

Comment (Shreyaschorge) [inline src/neynar-api/v1/client.ts:77]: @eurbs You can remove this line, it'll work regardless of this, cuz we are assigning apiKey when we instantiate a config object

```
new Configuration({
      basePath: basePath ? `${basePath}/v1` : BASE_PATH,
      apiKey: apiKey,
    })
```

**Review by Shreyaschorge — CHANGES_REQUESTED** [review]

**Review by Shreyaschorge — APPROVED** [review]
LGTM

**Pull request merged**